### PR TITLE
[FSSS-163] Feat: Style IconButton

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?
- Styles `IconButton` component
- Uses `IconButton` over `Button` component when applicable

## How it works? 
- Visually doesn't' change much. It helps us to keep icon button style on the component.

|Filter - close button|Menu| Mini cart - close button|
|-|-|-|
Filter - close button|Menu|Mini cart

## References
- [FSSS-163 Create IconButton](https://vtex-dev.atlassian.net/jira/software/projects/FSSS/boards/632?selectedIssue=FSSS-163)

### CODE REVIEW

**Review by eduardoformiga — APPROVED** [review]
Good Job! 👏

File: src/components/ui/IconButton/IconButton.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+interface IconButtonProps {
+  classes?: string
+}
+
+type Props = IconButtonProps & UIIconButtonProps
+
+function IconButton({ icon, classes, ...props }: Props) {
+  return (
+    <UIIconButton className={`icon-button ${classes}`} icon={icon} {...props} />
```
Comment (filipewl) [inline src/components/ui/IconButton/IconButton.tsx:15]: We normally call the rest params this way.

```suggestion
function IconButton({ icon, classes, ...otherProps }: Props) {
  return (
    <UIIconButton className={`icon-button ${classes}`} icon={icon} {...otherProps} />
```

File: src/components/cart/CartSidebar/cart-sidebar.scss [hunk 2]
```diff
@@ -30,12 +30,8 @@
...
   background-color: var(--bg-neutral-lightest);
 
-  button {
-    position: absolute;
-    right: calc(-1 * var(--page-padding-phone));
-    margin-right: var(--space-0);
-    color: var(--color-text);
-    background-color: transparent;
+  .cart-sidebar__button {
+    margin-right: calc(-1 * var(--space-1));
```
Comment (filipewl) [inline src/components/cart/CartSidebar/cart-sidebar.scss:34]: Is this negative margin intentional?

https://user-images.githubusercontent.com/381395/151890587-61f1586c-b5e4-4d51-851e-70e68709af51.mp4

File: src/components/ui/IconButton/icon-button.scss [hunk 3]
```diff
@@ -0,0 +1,31 @@
+@import "../../../styles/utilities";
+
+.icon-button[data-store-icon-button] {
```
Comment (filipewl) [inline src/components/ui/IconButton/icon-button.scss:3]: Do we need `.icon-button` or can we go just with `[data-store-icon-button]`?

**Review by filipewl — APPROVED** [review]
LGTM but left some questions!

File: src/components/ui/IconButton/icon-button.scss (hunk 3 above)
Comment (hellofanny) [reply src/components/ui/IconButton/icon-button.scss:3]: actually, we can go just with [data-store-icon-button].

**Comment by filipewl** [comment]
Merging `master` here because I think it will fix (thanks to #298) the [Build error](https://github.com/vtex-sites/base.store/runs/5028926444) we had in the last 2 empty commits.

CleanShot 2022-02-01 at 20 11 10

**Comment by hellofanny** [comment]
> Merging `master` here because I think it will fix (thanks to #298) the [Build error](https://github.com/vtex-sites/base.store/runs/5028926444) we had in the last 2 empty commits.
> 
> CleanShot 2022-02-01 at 20 11 10

I thought the same thing. I'll wait a little bit :)

**Pull request merged**